sklearn/decomposition/sparse_pca.py

```diff
@@ -231,6 +245,57 @@ def transform(self, X, ridge_alpha='deprecated'):
 
         return U
 
+    def _get_explained_variance(self, X):
```

Comment:
This seems to be a generalised approach? Perhaps pull this out into a module-level function taking X and components